server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java

```diff
@@ -351,6 +325,101 @@ public void onTimeout(TimeValue timeout) {
                 }
             }, statePredicate);
         }
+
+        private ActionListener<Response> getDelegateForLocalExecute(ClusterState clusterState) {
+            return ActionListener.delegateResponse(listener, (delegatedListener, t) -> {
+                if (t instanceof FailedToCommitClusterStateException || t instanceof NotClusterManagerException) {
+                    logger.debug(
+                        () -> new ParameterizedMessage(
+                            "master could not publish cluster state or " + "stepped down before publishing action [{}], scheduling a retry",
+                            actionName
+                        ),
+                        t
+                    );
+
+                    retryOnMasterChange(clusterState, t);
+                } else {
+                    delegatedListener.onFailure(t);
+                }
+            });
+        }
+
+        private void execOnClusterManagerOnTermMismatch(DiscoveryNode clusterManagerNode, ClusterState clusterState) {
+            transportService.sendRequest(
+                clusterManagerNode,
+                org.opensearch.action.admin.cluster.state.term.GetTermVersionAction.NAME,
+                new org.opensearch.action.admin.cluster.state.term.GetTermVersionRequest(),
+                new TransportResponseHandler<org.opensearch.action.admin.cluster.state.term.GetTermVersionResponse>() {
+                    @Override
+                    public void handleResponse(org.opensearch.action.admin.cluster.state.term.GetTermVersionResponse response) {
+                        boolean shouldExecuteOnClusterManger = !response.matches(clusterState);
+                        if (shouldExecuteOnClusterManger) {
+                            executeOnClusterManager(clusterManagerNode, clusterState);
+                        } else {
+                            Runnable runTask = ActionRunnable.wrap(
+                                getDelegateForLocalExecute(clusterState),
+                                l -> clusterManagerOperation(task, request, clusterState, l)
+                            );
+                            threadPool.executor(executor).execute(runTask);
+                        }
+                    }
+
+                    @Override
+                    public void handleException(TransportException exp) {
+                        handleTransportException(clusterManagerNode, clusterState, exp);
+                    }
+
+                    @Override
+                    public String executor() {
+                        return ThreadPool.Names.SAME;
+                    }
+
+                    @Override
+                    public org.opensearch.action.admin.cluster.state.term.GetTermVersionResponse read(StreamInput in) throws IOException {
```

Comment:
Can't we write it in import ? why full path is needed ?

Reply:
Looks to be IDE setting which replaced this. corrected it.